evalml/tests/component_tests/test_varmax_regressor.py

```diff
@@ -0,0 +1,393 @@
+    assert len(y_pred) == 10
+    assert y_pred.index.equals(X_test.index)
+
+
+@pytest.mark.parametrize("freq_num", ["1", "2"])
+@pytest.mark.parametrize("freq_str", ["T", "M", "Y"])
+def test_different_time_units_out_of_sample(
+    freq_str,
+    freq_num,
+    sktime_varmax,
+    forecasting,
+):
+    from sktime.forecasting.base import ForecastingHorizon
+    from sktime.forecasting.varmax import VARMAX
+
+    datetime_ = pd.date_range("1/1/1870", periods=20, freq=freq_num + freq_str)
+
+    X = pd.DataFrame(
+        range(20),
+        index=datetime_,
+    )
+    y = pd.DataFrame(
+        {
+            "target_0": pd.Series(
+                np.sin(np.linspace(-8 * np.pi, 8 * np.pi, 20)),
+                index=datetime_,
+            ),
+            "target_1": pd.Series(
+                np.sin(np.linspace(-4 * np.pi, 4 * np.pi, 20)),
+                index=datetime_,
+            ),
+        },
+    )
+
+    fh_ = ForecastingHorizon([i + 1 for i in range(len(y[15:]))], is_relative=True)
+
+    a_clf = VARMAX(maxiter=10)
+    clf = a_clf.fit(X=X[:15], y=y[:15])
+    y_pred_sk = clf.predict(fh=fh_, X=X[15:])
+
+    m_clf = VARMAXRegressor(series_id="series_id")
+    m_clf.fit(X=X[:15], y=y[:15])
+    y_pred = m_clf.predict(X=X[15:])
+
+    np.testing.assert_almost_equal(y_pred_sk.values, y_pred.values)
+    assert y_pred.index.equals(X[15:].index)
+
+
+@patch("sktime.forecasting.varmax.VARMAX.fit")
+def test_varmax_supports_boolean_features(mock_fit):
+    X = pd.DataFrame({"dates": pd.date_range("2021-01-01", periods=10)})
+    X.ww.init()
+    X.ww["bool_1"] = (
+        pd.Series([True, False])
+        .sample(n=10, replace=True, random_state=0)
+        .reset_index(drop=True)
+    )
+    X.ww["bool_2"] = (
+        pd.Series([True, False])
+        .sample(n=10, replace=True, random_state=1)
+        .reset_index(drop=True)
+    )
+    y = pd.DataFrame({"target_1": np.random.rand(10), "target_2": np.random.rand(10)})
+
+    vx = VARMAXRegressor(time_index="dates", series_id="series_id")
+    vx.fit(X, y)
+
+    pd.testing.assert_series_equal(
+        mock_fit.call_args[1]["X"]["bool_1"],
+        X["bool_1"].astype(float),
+    )
+    pd.testing.assert_series_equal(
+        mock_fit.call_args[1]["X"]["bool_2"],
+        X["bool_2"].astype(float),
+    )
+
+
+def test_varmax_boolean_features_no_error():
+    X = pd.DataFrame({"dates": pd.date_range("2021-01-01", periods=10)})
+    X.ww.init()
+    X.ww["bool_1"] = (
+        pd.Series([True, False])
+        .sample(n=10, replace=True, random_state=0)
+        .reset_index(drop=True)
+    )
+    X.ww["bool_2"] = (
+        pd.Series([True, False])
+        .sample(n=10, replace=True, random_state=1)
+        .reset_index(drop=True)
+    )
+    y = pd.DataFrame({"target_1": np.random.rand(10), "target_2": np.random.rand(10)})
+
+    vx = VARMAXRegressor(time_index="dates", series_id="series_id")
+    vx.fit(X, y)
+    preds = vx.predict(X)
+    assert all(preds.isna().eq(False))
```

Comment:
This test is identical to the previous one save the mocking of fit and the very last line.... that's a lot of repeated code. Can we pull out the data into a new fixture/add the possibility of boolean to an existing fixture, or combine these two tests, or something? 

Really, as I'm thinking about it, I don't think we need the first test if we have the second - it doesn't really matter that VARMAX converts bools to floats so long as at the end of the day, it handles them. Right?